mobile/src/main/java/ch/epfl/sdp/mobile/ui/features/profile/ProfileStateScreen.kt

```diff
@@ -0,0 +1,174 @@
+  //  GamesInfo(numPastGames = 42, numPuzzles = 10)
+  ProfileScreen(state = state)
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun ProfileScreen(state: ProfileState, modifier: Modifier = Modifier) {
+  val strings = LocalLocalizedStrings.current
+  LazyColumn(
+      verticalArrangement = Arrangement.Top,
+      horizontalAlignment = Alignment.CenterHorizontally,
+      modifier = modifier) {
+    item { ProfileHeader(state) }
+    stickyHeader {
+      GamesInfo(numPastGames = state.numberOfGames, numPuzzles = state.numberOfPuzzles)
+    }
+    items(state.matches) { match ->
+      val title = strings.profileMatchTitle(match.adv)
+      val subtitle = strings.profileMatchInfo(match.matchResult, match.cause, match.numberOfMoves)
+      Match(title, subtitle, modifier = modifier)
+    }
+  }
+}
+
+@Composable
+fun ProfileHeader(state: ProfileState, modifier: Modifier = Modifier) {
+  Column(
+      modifier = modifier,
+      horizontalAlignment = Alignment.CenterHorizontally,
+  ) {
+    Spacer(Modifier.height(16.dp))
+    ProfilePicture(state)
+    Spacer(Modifier.height(16.dp))
+    Text(state.name, style = MaterialTheme.typography.h5)
+    Text(state.email, style = MaterialTheme.typography.subtitle2)
+    Spacer(Modifier.height(16.dp))
+    SettingsButton(onClick = state::onSettingsClick)
+    Spacer(Modifier.height(16.dp))
+  }
+}
+
+@Composable
+fun ProfilePicture(
+    state: ProfileState,
+    modifier: Modifier = Modifier,
+) {
+  Box(
+      modifier =
+          modifier.size(118.dp).background(state.backgroundColor.getColorForProfile(), CircleShape),
+      contentAlignment = Alignment.Center,
+  ) {
+    Text(state.emoji, style = MaterialTheme.typography.h3)
+    IconButton(
+        onClick = state::onEditClick,
+        modifier =
+            Modifier.align(Alignment.BottomEnd)
+                .background(MaterialTheme.colors.surface, CircleShape)
+                .border(2.dp, MaterialTheme.colors.primary, CircleShape)
+                .shadow(2.dp, CircleShape)
+                .size(40.dp),
+    ) { Icon(Icons.Default.Edit, "Edit profile icon") }
+  }
+}
+
+@Composable
+fun SettingsButton(onClick: () -> Unit, modifier: Modifier = Modifier) {
+  val strings = LocalLocalizedStrings.current
+  Button(
+      onClick = onClick,
+      shape = CircleShape,
+      contentPadding = PaddingValues(horizontal = 24.dp, vertical = 12.dp),
+      modifier = modifier) {
+    Icon(Icons.Default.Settings, null)
+    Spacer(modifier = Modifier.width(8.dp))
+    Text(strings.profileSettings)
+  }
+}
+
+@Composable
+fun infoComponent(text: String, num: Int, modifier: Modifier = Modifier) {
+  // Past games component
+  Column(
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.Center,
+      modifier = modifier) {
+    Text(text = text, modifier = modifier)
+    Text(
+        text = "$num",
+        style = TextStyle(textDecoration = TextDecoration.Underline),

```

Comment:
You may want to actually use a custom modifier, rather than a `TextDecoration` (which does not let you control the width of the stroke). Something like this might do the trick :

```
/**
 * A [Modifier] which can be used to draw a border at the bottom of the composable, which fits the
 * full width of the composable.
 *
 * @param width the width of the stroke.
 * @param color the [Color] with which the stroke is drawn.
 */
private fun Modifier.borderBottom(
    width: Dp,
    color: Color,
): Modifier = drawBehind {
  val start = Offset(0f, size.height)
  val end = Offset(size.width, size.height)
  drawLine(color, start, end, width.toPx(), StrokeCap.Round)
}
```

You could then use it as a `Modifier` on the bottom text.